packages/core/strapi/lib/Strapi.js

```diff
@@ -142,7 +143,7 @@ class Strapi {
   }
 
   get customFields() {
-    return this.container.get('custom-fields');
+    return createCustomFields(this);
```

Comment:
this will recreate the service everytime.  You will want to do this in the constructor instead here.